src/refs.c

```diff
@@ -624,13 +624,23 @@ static int update_wt_heads(git_repository *repo, const char *path, void *payload
 	rename_cb_data *data = (rename_cb_data *) payload;
 	git_reference *head;
 	char *gitdir = NULL;
-	int error = 0;
+	int error;
+
+	if ((error = git_reference__read_head(&head, repo, path)) < 0) {
```

Comment:
If `git_reference__read_head` fails, `head` will never be initialized, and we'll call `git__free` on it.  (`read_head` also does not set it to `NULL`.)  Can we initialize it to `NULL` before this?